### PR TITLE
Fix test case explain_format about Extra Text

### PR DESCRIPTION
**Pull request by yaowangm**
By https://github.com/greenplum-db/gpdb/pull/15917, some code was added to test case explain_format to check the EXPLAIN ANALYZE output for hash agg. However, it failed occasionally because there might be an additional line of "Extra Text" like this:
```
@@ -871,6 +871,8 @@
                     Group Key: share0_ref2.b
                     Extra Text: (seg0)   hash table(s): ###; ### groups total in ### batches, ### spill partitions; disk usage: ###KB; chain length ###.## avg, ### max; using ## of ### buckets; total ### expansions.
+                    Extra Text: (seg0)   hash table(s): ###; ### groups total in ### batches, ### spill partitions; disk usage: ###KB; chain length ###.## avg, ### max; using ## of ### buckets; total ### expansions.
+
                     ->  Redistribute Motion 3:3  (slice2; segments: 3) (actual time=##.###..##.### rows=# loops=#)
                           Hash Key: share0_ref2.b
                           ->  Result (actual time=##.###..##.### rows=# loops=#)
```
The output is from the code in `cdbexplain_depositStatsToNode()`:
```
1121         /*
1122          * For the worker which had the highest peak executor memory usage
1123          * overall across the whole slice, we'll report the extra message text
1124          * from all of the nodes in the slice.  But only if that worker stands
1125          * out more than 5% above the average.
1126          */
1127         if (peakmemused.agg.vmax > 1.05 * cdbexplain_agg_avg(&peakmemused.agg))
1128             cdbexplain_depStatAcc_saveText(&peakmemused, ctx->extratextbuf, &saved);
```
It means, one than more line might be printed if the condition is qualified, i.e. it works as design.

The fix is to use an existing UDF `test_util.get_explain_output()` to check the result and avoid the failure.

### CODE REVIEW

File: src/test/regress/expected/explain_format.out [hunk 1]
```diff
@@ -920,56 +920,34 @@ CREATE TABLE test_hashagg_spill AS
...
-Memory wanted:  1640kB
-Execution Time: 22.346 ms
-(17 rows)
+WITH query_plan (et) AS
+(
+  SELECT test_util.get_explain_output(
+    'SELECT a, COUNT(DISTINCT b) AS b FROM test_src_tbl GROUP BY a',
+    false)
+)
+SELECT BTRIM(et) as epln_info FROM query_plan WHERE et like '%Extra Text%' limit 2;
```
Comment (adam8157) [inline src/test/regress/expected/explain_format.out:929]: `epln` is not a common abbreviation, not a big deal anyway.

**Review by adam8157 — COMMENTED** [review]
The failed difference you mentioned in the PR message is:

```
@@ -871,6 +871,8 @@
                     Group Key: share0_ref2.b
                     Extra Text: (seg0)   hash table(s): ###; ### groups total in ### batches, ### spill partitions; disk usage: ###KB; chain length ###.## avg, ### max; using ## of ### buckets; total ### expansions.
+                    Extra Text: (seg0)   hash table(s): ###; ### groups total in ### batches, ### spill partitions; disk usage: ###KB; chain length ###.## avg, ### max; using ## of ### buckets; total ### expansions.
+
                     ->  Redistribute Motion 3:3  (slice2; segments: 3) (actual time=##.###..##.### rows=# loops=#)
                           Hash Key: share0_ref2.b
                           ->  Result (actual time=##.###..##.### rows=# loops=#)
```

Both lines are from `seg0`, but the explanation in `cdbexplain_depositStatsToNode()` expects more than one extra text line from different nodes/workers, correct? Seems it doesn't match.

**Comment by yaowangm** [comment]
> The failed difference you mentioned in the PR message is:
> 
> ```
> @@ -871,6 +871,8 @@
>                      Group Key: share0_ref2.b
>                      Extra Text: (seg0)   hash table(s): ###; ### groups total in ### batches, ### spill partitions; disk usage: ###KB; chain length ###.## avg, ### max; using ## of ### buckets; total ### expansions.
> +                    Extra Text: (seg0)   hash table(s): ###; ### groups total in ### batches, ### spill partitions; disk usage: ###KB; chain length ###.## avg, ### max; using ## of ### buckets; total ### expansions.
> +
>                      ->  Redistribute Motion 3:3  (slice2; segments: 3) (actual time=##.###..##.### rows=# loops=#)
>                            Hash Key: share0_ref2.b
>                            ->  Result (actual time=##.###..##.### rows=# loops=#)
> ```
> 
> Both lines are from `seg0`, but the explanation in `cdbexplain_depositStatsToNode()` expects more than one extra text line from different nodes/workers, correct? Seems it doesn't match.

In most cases, `cdbexplain_depositStatsToNode()` prints only one lines for a hash agg executor. And there are two hash agg executors in the plan, so two lines of "Extra Text". That's why you saw two lines from `seg0` which were from different executors. Like this:
```
Executor 1:
Extra Text: (seg0)   hash table(s): ###;
Executor 2:
Extra Text: (seg0)   hash table(s): ###;
```
In some cases, `cdbexplain_depositStatsToNode()` prints two lines for a hash agg executor. So there are four lines of "Extra Text": each executors has two lines from different segs, like this:
```
Executor 1:
Extra Text: (seg0)   hash table(s): ###;
Extra Text: (seg1)   hash table(s): ###;
Executor 2:
Extra Text: (seg0)   hash table(s): ###;
Extra Text: (seg1)   hash table(s): ###;
```
The test case checks only the first two lines, no matter which executor they belong to, to avoid the issue.

File: src/test/regress/expected/explain_format.out (hunk 1 above)
Comment (yaowangm) [reply src/test/regress/expected/explain_format.out]: Chanted to `explain_info`

**Review by adam8157 — APPROVED** [review]
I got it, there are two seg0 extra text lines because we replaced in cases...

```
-- m/Extra Text: \(seg\d+\)   hash table\(s\): \d+; \d+ groups total in \d+ batches, \d+ spill partitions; disk usage: \d+KB; chain length \d+.\d+ avg, \d+ max; using \d+ of \d+ buckets; total \d+ expansions./
-- s/Extra Text: \(seg\d+\)   hash table\(s\): \d+; \d+ groups total in \d+ batches, \d+ spill partitions; disk usage: \d+KB; chain length \d+.\d+ avg, \d+ max; using \d+ of \d+ buckets; total \d+ expansions./Extra Text: (seg0)   hash table(s): ###; ### groups total in ### batches, ### spill partitions; disk usage: ###KB; chain length ###.## avg, ### max; using ## of ### buckets; total ### expansions./
```

**Pull request merged**